tests/local_cache_test.py

```diff
@@ -40,6 +40,15 @@ def test_add(self) -> None:
         self.cache.add(text)
         self.assertIn(text, self.cache.data.texts)
 
+    @requires_api_key("OPENAI_API_KEY")
+    def test_add_non_utf8_str(self) -> None:
```

Comment:
Does this require the api key to work?

Reply:
@ntindle ,
Please advise, In the first time I pulled this code it didn't include `@requires_api_key("OPENAI_API_KEY")`
Then the master was changed and I had a conflict due to the addition of @requires_api_key("OPENAI_API_KEY") to the rest of the functions. So I added it to my function as well. Since it is the same as the others. 
So why do you think this function is different?

antway I am rebasing master to see if it solves the tests